pkg/transport/limited_conn_listener.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transport
+
+import (
+	"net"
+	"time"
+
+	"github.com/coreos/etcd/pkg/runtime"
+)
+
+type LimitedConnListener struct {
+	net.Listener
+	RuntimeFDLimit uint64
+}
+
+func (l *LimitedConnListener) Accept() (net.Conn, error) {
+	for {
```

Comment:
The Accept should always return a valid connection and nil error, or some error. And http.Server.Serve() uses this assumption: https://golang.org/src/net/http/server.go#L1724

Accept is supposed to wait for the next call and return a general conn, so it is reasonable for it to be blocked.